week3/src/pages/Game.jsx

```diff
@@ -0,0 +1,170 @@
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import {
+  img72Inch,
+  img8Inch,
+  imgCopyAndPaste,
+  imgGoogling,
+  imgKeyboard,
+  imgLeftHand,
+  imgTouchPad,
+  imgVoice,
+} from 'assets';
+
+function Game() {
+  const navigate = useNavigate();
+  const [matchIndex, setMatchIndex] = useState(0);
+  const [matchItemList, setMatchItemList] = useState([
+    { id: 1, img: imgGoogling, caption: '구글링 없이 개발하기' },
+    { id: 2, img: imgCopyAndPaste, caption: '복붙 없이 개발하기' },
+    { id: 3, img: img72Inch, caption: '1k 해상도 72인치 모니터로 코딩하기' },
+    { id: 4, img: img8Inch, caption: '4k 해상도 8인치 모니터로 코딩하기' },
+    { id: 5, img: imgKeyboard, caption: '천지인 자판으로 코딩하기' },
+    { id: 6, img: imgVoice, caption: '음성인식으로 코딩하기' },
+    { id: 7, img: imgLeftHand, caption: '왼손으로만 마우스 쓰기' },
+    { id: 8, img: imgTouchPad, caption: '노트북 터치 패드만 쓰기' },
+  ]);
+  const [nextMatchItemList, setNextMatchItemList] = useState([]);
+
+  const makeMatchList = (matchItemList) => {
+    const matchList = [];
+    let match = [];
+    matchItemList.forEach((item, i) => {
+      if (i % 2 === 0) {
+        match = [item];
+        return;
+      }
+      matchList.push([...match, item]);
+    });
+    return matchList;
+  };
+
```

Comment:
array를 let으로 선언해서  주소를 계속 바꾸는 것은 좋지 않다고 들었어요 ..!

```suggestion
  const makeMatchList = (matchItemList) => {
    const matchList = [];
    for (const i=0; i < matchItemList.length; i+=2) {
      matchList.push(matchItemList.slice(i, i+2);
    };
    return matchList;
  };

```
요런 식으로 for문과 slice함수를 써서 구현해볼 수 있을 것 같습니다!

Reply:
가독성도 좋아지고, 길이도 짧아지고, 훨씬 맘에 드는 코드네요! 감사합니당👏